src/Layers/xrRenderGL/glHW.cpp

```diff
@@ -86,8 +85,8 @@ void CHW::CreateDevice(SDL_Window* hWnd)
     SDL_SetWindowDisplayMode(m_window, &mode);
 
     // Create the context
-    m_context = SDL_GL_CreateContext(m_window);
-    if (m_context == nullptr)
+    m_context = nullptr; // Clean up first
+    if (! (m_context = SDL_GL_CreateContext(m_window)));
```

Comment:
1. The `m_context` variable isn't passed to `SDL_GL_CreateContext` function and this function has a return meaning that it will always write to `m_context`, so there's no need to make it `nullptr`.
2. Moreover, the `;` effectively disabled the if and made the error handling code below always enabled, meaning that it would always write `! Could not create drawing context:` to the log and return even if the context has been successfully created.
That also means all the code below like `MakeContextCurrent(IRender::PrimaryContext)` won't be called at all.

I suggest to restore the original code:
```suggestion
    m_context = SDL_GL_CreateContext(m_window);
    if (m_context == nullptr)
```

Reply:
Look at this example:
```cpp
if (condition)
    ; // do nothing
{
    // this code is NOT the part of the if,
    // it will always be executed.
    // it's just a free standing scope
    int i = 0;
    return;
}

// this code never will be executed
int k = 0;
```

Reply:
A stupid mistake of my part, that ";" shouldn't be there in the first place.
Nevertheless, I understand that mi "security" proposal shouldn't make much difference if a segfault occurs.

Reply:
It doesn't add anything, because `SDL_GL_CreateContext` does always write to `m_context`, so this change makes code less readable for no purpose.